src/poetry/console/commands/check.py

```diff
@@ -158,7 +158,7 @@ def handle(self) -> int:
         if self.poetry.locker.is_locked() and not self.poetry.locker.is_fresh():
             check_result["errors"] += [
                 "pyproject.toml changed significantly since poetry.lock was last generated. "
-                "Run `poetry lock [--no-update]` to fix the lock file."
+                "Run `poetry lock` to fix the lock file."
```

Comment:
Or alternatively

```suggestion
                "Run `poetry lock [--regenerate]` to fix the lock file."
```

Reply:
I think it is sufficient to mention only the option that will result in the fewest changes.